Java/core/src/main/java/com/amazon/randomcutforest/sampler/AbstractStreamSampler.java

```diff
@@ -158,4 +184,91 @@ public void setMostRecentTimeDecayUpdate(long index) {
             throw new IllegalArgumentException("Unsupported configuration setting: " + name);
         }
     }
+
+    /**
+     * @return the number of points contained by the sampler when full.
+     */
+    @Override
+    public int getCapacity() {
+        return capacity;
+    }
+
+    public double getInitialAcceptFraction() {
+        return initialAcceptFraction;
+    }
+
+    public long getRandomSeed() {
+        return random.randomSeed;
+    }
+
+    protected class ReplayableRandom {
+        long randomSeed;
+        Random testRandom;
+
+        ReplayableRandom(long randomSeed) {
+            this.randomSeed = randomSeed;
+        }
+
+        ReplayableRandom(Random random) {
+            this.testRandom = random;
+        }
+
+        double nextDouble() {
```

Comment:
We should put a note somewhere that the sampler is not concurrency safe. There's already a disclaimer in the README, but we might want to make it more expansive and say something like "assume that classes in this library are not concurrency safe unless explicitly state otherwise in the documentation", and then maybe some words about how we support parallelism by synchronizing access to shared state.

Reply:
Agree.